helpers/argumentParsers.go

```diff
@@ -1708,45 +1707,127 @@ func ParseGetSocketOption(rawValue uint64) (SocketOptionArgument, error) {
+		BPFProgTypeTracing:               "BPF_PROG_TYPE_TRACING",
+		BPFProgTypeStructOps:             "BPF_PROG_TYPE_STRUCT_OPS",
+		BPFProgTypeExt:                   "BPF_PROG_TYPE_EXT",
+		BPFProgTypeLsm:                   "BPF_PROG_TYPE_LSM",
+		BPFProgTypeSkLookup:              "BPF_PROG_TYPE_SK_LOOKUP",
+		BPFProgTypeSyscall:               "BPF_PROG_TYPE_SYSCALL",
+	}
+	str = x[b]
+	if str == "" {
+		str = BPFProgTypeUnspec.String()
+	}
+	return str
+}
+
+var bpfProgTypeMap = map[uint64]BPFProgType{
+	BPFProgTypeUnspec.Value():                BPFProgTypeUnspec,
+	BPFProgTypeSocketFilter.Value():          BPFProgTypeSocketFilter,
+	BPFProgTypeKprobe.Value():                BPFProgTypeKprobe,
+	BPFProgTypeSchedCls.Value():              BPFProgTypeSchedCls,
+	BPFProgTypeSchedAct.Value():              BPFProgTypeSchedAct,
+	BPFProgTypeTracepoint.Value():            BPFProgTypeTracepoint,
+	BPFProgTypeXdp.Value():                   BPFProgTypeXdp,
+	BPFProgTypePerfEvent.Value():             BPFProgTypePerfEvent,
+	BPFProgTypeCgroupSkb.Value():             BPFProgTypeCgroupSkb,
+	BPFProgTypeCgroupSock.Value():            BPFProgTypeCgroupSock,
+	BPFProgTypeLwtIn.Value():                 BPFProgTypeLwtIn,
+	BPFProgTypeLwtOut.Value():                BPFProgTypeLwtOut,
+	BPFProgTypeLwtXmit.Value():               BPFProgTypeLwtXmit,
+	BPFProgTypeSockOps.Value():               BPFProgTypeSockOps,
+	BPFProgTypeSkSkb.Value():                 BPFProgTypeSkSkb,
+	BPFProgTypeCgroupDevice.Value():          BPFProgTypeCgroupDevice,
+	BPFProgTypeSkMsg.Value():                 BPFProgTypeSkMsg,
+	BPFProgTypeRawTracepoint.Value():         BPFProgTypeRawTracepoint,
+	BPFProgTypeCgroupSockAddr.Value():        BPFProgTypeCgroupSockAddr,
+	BPFProgTypeLwtSeg6Local.Value():          BPFProgTypeLwtSeg6Local,
+	BPFProgTypeLircMode2.Value():             BPFProgTypeLircMode2,
+	BPFProgTypeSkReuseport.Value():           BPFProgTypeSkReuseport,
+	BPFProgTypeFlowDissector.Value():         BPFProgTypeFlowDissector,
+	BPFProgTypeCgroupSysctl.Value():          BPFProgTypeCgroupSysctl,
+	BPFProgTypeRawTracepointWritable.Value(): BPFProgTypeRawTracepointWritable,
+	BPFProgTypeCgroupSockopt.Value():         BPFProgTypeCgroupSockopt,
+	BPFProgTypeTracing.Value():               BPFProgTypeTracing,
+	BPFProgTypeStructOps.Value():             BPFProgTypeStructOps,
+	BPFProgTypeExt.Value():                   BPFProgTypeExt,
+	BPFProgTypeLsm.Value():                   BPFProgTypeLsm,
+	BPFProgTypeSkLookup.Value():              BPFProgTypeSkLookup,
+	BPFProgTypeSyscall.Value():               BPFProgTypeSyscall,
 }
 
-func ParseBPFProgType(rawValue uint64) (libbpfgo.BPFProgType, error) {
+func ParseBPFProgType(rawValue uint64) (BPFProgType, error) {
 	v, ok := bpfProgTypeMap[rawValue]
 	if !ok {
-		return libbpfgo.BPFProgType(0), fmt.Errorf("not a valid argument: %d", rawValue)
+		return BPFProgType(0), fmt.Errorf("not a valid argument: %d", rawValue)
```

Comment:
What about instead of another map we implement only one following this struct logic:

```go
const (
	BPFProgTypeUnspec uint64 = iota
	BPFProgTypeSocketFilter
)

type BPFProgType struct {
	name  string
	value uint64
}

var bpfProgTypeMap = map[uint64]BPFProgType{
	BPFProgTypeUnspec: {
		name:  "BPF_PROG_TYPE_UNSPEC",
		value: BPFProgTypeUnspec,
	},
	BPFProgTypeSocketFilter: {
		name: "BPF_PROG_TYPE_SOCKET_FILTER",
		value: BPFProgTypeSocketFilter,
	},
}

func (b BPFProgType) String() string {
	return b.name
}

func (b BPFProgType) Value() uint64 {
	return b.value
}
```

Reply:
Oh, I think I got it. Keeping the Name string is not required.

Said that, why not just to make a conversion here and get rid of the the map that converts from uint64 to uint32?

```go
func ParseBPFProgType(rawValue uint64) (BPFProgType, error) {
	v, ok := bpfProgTypeMap[uint32(rawValue)]
```

Reply:
I see what you mean, I believe uint64 was purposefully used as a parsing parameter because thats how the value is read from raw events, not 100% sure though.